src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1078,7 +1078,7 @@ private static FunctionMetadata ParseFunctionMetadata(string functionName, JObje
             return functionMetadata;
         }
 
-        public static Collection<FunctionMetadata> ReadFunctionsMetadata(IEnumerable<string> functionDirectories, ILogger logger, Dictionary<string, Collection<string>> functionErrors, ScriptSettingsManager settingsManager = null, IEnumerable<string> functionWhitelist = null)
+        public static Collection<FunctionMetadata> ReadFunctionsMetadata(IEnumerable<string> functionDirectories, ILogger logger, Dictionary<string, Collection<string>> functionErrors, ScriptSettingsManager settingsManager = null, IEnumerable<string> functionWhitelist = null, IFileSystem fileSystem = null)
```

Comment:
methods in this file are inconsistent in their use of either a passed in IFileSystem defaulting to null and use of `System.IO.File` and `System.IO.Directory` if the `IFileSystem` is null **vs** using `FileUtility` with a global `Instance` of `IFileSystem`. I think they should all be unified to just use `FileUtility`, but I'm not sure if you're trying to move away from `FileUtility` or to it.

Reply:
Yes things are inconsistent, mainly because FileUtility was added late for mocking, and was only used on targeted new APIs.